### DISCUSSION TITLE
Multiple temperature and humidity sensors

### DISCUSSION THREAD

**Discussion by Schwenkner** [t0bst4r/home-assistant-matter-hub#295] [Q&A] [answered]
I have a temperature/humidity sensor in almost every room. Just tried the matter hub with theses sensors. Therefor my solution was to name the sensor temperature entities with their room names. Thus I can ask "What temperature is in 'roomname'". This works. But how can I add humidity then? Naming the humidity entity also with the room name seems no solution since the name is also assigned for the temperature. Naming the device like "roomname temperature", then I get the answer "The temperatur of roomname temperature is ...".

How could this be achieved?

**Comment by danielbrunt57** [comment]
> How could this be achieved?

It can't.

Amazon/Alexa does not support reporting humidity, even when using their own "_Amazon Indoor Air Quality Monitor_" which I have attached to the group "Living Room".
Q: "Alexa, what's the humidity in the living room"
A: "That's not supported yet"

Temperature via the AIAQM though works...
Q: "Alexa, what's the temperature in the living room"
A; "The average temperature of living room is 20.8 degrees"

**Comment by danielbrunt57** [comment]
I found this **_reddit_** post from two years ago: [Still Waiting For Humidity & Air Quality Compatibility](https://www.reddit.com/r/amazonecho/comments/14x0ie3/still_waiting_for_humidity_air_quality/)
I renamed my AQM to "Living Room Air" and I can now ask:
Q: "Alexa, what is the living room air temperature"
A: "the living room air temperature is 20.5 degrees"

Q: "Alexa, what is the living room air humidity"
A: "the indoor humidity is 42%"

I have exposed two HA humidity sensors: _Laundry Room Humidity_ & _Garage Humidity_ that I renamed _Laundry Room Air_ & _Garage Air_ but when I ask "Alexa, what is the garage air humidity" like I did with the AQM, she still answers "that's not supported yet"...

**Comment by knuti1960** [comment]
I therefore use scripts for the humidity that I can call up via a routine. "Alexa, what's the humidity in the bathroom" activates the following script:

`alias: Luftfeuchtigkeitsabrage Badezimmer
sequence:
  - data: {}
    enabled: true
    action: alexa_media.update_last_called
  - delay:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 2
  - data:
      target: "{{ states('sensor.last_alexa') }}"
      message: >
        {% set alexa_text = [ "Die Luftfeuchtigkeit im Badezimmer beträgt ",
                              "Im Badezimmer sind es ", 
                              "Im Moment herscht im Badezimmer eine Luftfeuchte von " ] | random %}
        {{ '{} {} Prozent.'.format( alexa_text,
        states('sensor.klima_badezimmer_humidity')) }}
      data:
        type: tts
        method: speak
    action: notify.alexa_media
mode: single
icon: mdi:percent
description: ""`

The disadvantage here is only Alexa's confirming "Okay" before the actual answer.
EDIT: Unfortunately, I am not able to post the script with the correct indentations. Therefore again as an attachment
[Script Luftfeuchtigkeitsabfrage.txt](https://github.com/user-attachments/files/18527103/Script.Luftfeuchtigkeitsabfrage.txt)

**Comment by danielbrunt57** [comment, marked as answer]
@knuti1960 Use three backquotes on their own line before and after the code block, or select the section of code and click `Code` in the menu bar (`<>`).

```
alias: Luftfeuchtigkeitsabrage Badezimmer
sequence:
  - data: {}
    enabled: true
    action: alexa_media.update_last_called
  - delay:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 2
  - data:
      target: "{{ states('sensor.last_alexa') }}"
      message: >
        {% set alexa_text = [ "Die Luftfeuchtigkeit im Badezimmer beträgt ",
                              "Im Badezimmer sind es ", 
                              "Im Moment herscht im Badezimmer eine Luftfeuchte von " ] | random %}
        {{ '{} {} Prozent.'.format( alexa_text,
        states('sensor.klima_badezimmer_humidity')) }}
      data:
        type: tts
        method: speak
    action: notify.alexa_media
mode: single
icon: mdi:percent
description: ""
```

**Comment by knuti1960** [comment]
Now I've got it, thank you!
